store/store_test.go

```diff
@@ -1707,6 +1925,52 @@ func (t *remoteRepoTestSuite) TestUbuntuStoreRepositoryAssertionRefreshesAuth(c
 	c.Check(refreshDischargeEndpointHit, Equals, true)
 }
 
+func (t *remoteRepoTestSuite) TestUbuntuStoreRepositoryAssertionRefreshesDeviceAuth(c *C) {
```

Comment:
@matiasb yes, can be in a follow up, but I think we need to separate:
- testing that all the needed places use doRequest
- testing what doRequest does in complicated cases

otherwise we get a bunch of tests that we need to change when either side changes and it can get a bit tedious, also lots of repetitive test code to read for someone coming in